tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -136,7 +149,150 @@ def product_level_df():
         {
             "timestamp": ["2000-01-01", "2000-01-02"] * 4,
             "segment": ["a"] * 2 + ["b"] * 2 + ["c"] * 2 + ["d"] * 2,
-            "target": [1, 2] + [10, 20] + [100, 200] + [1000, 2000],
+            "target": [1, 1] + [0, 1] + [3, 18] + [7, 2],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def product_level_df_w_nans():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03", "2000-01-04"] * 4,
+            "segment": ["a"] * 4 + ["b"] * 4 + ["c"] * 4 + ["d"] * 4,
+            "target": [None, 0, 1, 2] + [3, 4, 5, None] + [7, 8, None, 9] + [10, 11, 12, 13],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_df_w_nans():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03", "2000-01-04"] * 2,
+            "segment": ["X"] * 4 + ["Y"] * 4,
+            "target": [None, 4, 6, None] + [17, 19, None, 22],
```

Comment:
I am not sure that there should be nans, i guess the result should be "None + 3 = 3", but we can discuss it as it is not so obvious(we should come up with some convention for such cases)